### PR TITLE
HTTP Status code and handlers #39522

### PR DESCRIPTION
**Pull request by shashankbaranwal**
- Introduction to JSON.
- Introduction to body parser and its importance
- Why HTTP Status Code?
- Handling error and responses
 - Error Handler.
 - Success Handler.
 - Not Found Route.

### CODE REVIEW

File: src/libs/routes/notFoundRoute.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Request, Response, NextFunction } from 'express'
+
+export default (req, res, next) => {
+    next({
+        error : "Not Found",
+        code: 404
+    })
+}
```
Comment (namanpar) [inline src/libs/routes/notFoundRoute.ts:8]: Give proper name to function

File: src/libs/routes/errorHandler.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+export default ((err, req : Request, res : Response, next : NextFunction) => {
+    console.log(err);
+    res.json(
+        {
+            error : err.error,
+            status : err.code,
+            message : err.message || "Error",
+            timeStamp: new Date()
+        }
```
Comment (namanpar) [inline src/libs/routes/errorHandler.ts:11]: Give proper name to function

File: src/Server.ts [hunk 3]
```diff
@@ -1,31 +1,60 @@
...
+                    "error ": err.error,
+                    status : err.code,
+                    message : err. message || "Error",
+                    timeStamp: new Date()
+                 
+                }
+            )
+        });
+        return this;
+    }
```
Comment (namanpar) [inline src/Server.ts:44]: Already created errorHandler and notFoundHandler, remove unnecessary code

**Review by namanpar — CHANGES_REQUESTED** [review]

File: src/libs/routes/notFoundRoute.ts (hunk 1 above)
Comment (shashankbaranwal) [reply src/libs/routes/notFoundRoute.ts]: I forgot to add function but I will give proper function name  in this particular files.

File: src/libs/routes/errorHandler.ts (hunk 2 above)
Comment (shashankbaranwal) [reply src/libs/routes/errorHandler.ts]: I forgot to add function but I will give proper function name  in this particular files.

File: src/Server.ts (hunk 3 above)
Comment (shashankbaranwal) [reply src/Server.ts]: ok I will remove all unnecessary  code.

**Review by namanpar — APPROVED** [review]

**Pull request merged**